plugins/deta/api_key.go

```diff
@@ -0,0 +1,71 @@
+package deta
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://deta.space/docs/en"), // TODO: Replace with actual URL
+		ManagementURL: sdk.URL("https://deta.space"), // TODO: Replace with actual URL
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Deta.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 25,
+					Prefix: "bE53FsM6_", // TODO: Check if this is correct
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryDetaConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"DETA_API_KEY": fieldname.APIKey, // TODO: Check if this is correct
```

Comment:
Could you be pointing me please to any documentation about the `DETA_API_KEY` env var?

I found that `space` cli uses a `SPACE_ACCESS_TOKEN` instead: https://github.com/deta/space-cli#customizing-the-api-endpoint

Reply:
I also found a `DETA_PROJECT_KEY` but this seems to be for something else.

Finally, I found something about API Keys but still no reference to that env var: https://deta.space/docs/en/build/fundamentals/the-space-runtime/authentication#api-keys

Reply:
I think `DETA_API_KEY` may be the name that the template generated. So definitely worth having another look over this provisioner.

Reply:
> I think `DETA_API_KEY` may be the name that the `make new-plugin` command generated. So definitely worth having another look over this provisioner.

yes it is.